ocaml/xapi/xapi_xenops.ml

```diff
@@ -1131,6 +1131,8 @@ module MD = struct
           [Vm.Start]
       | `destroy ->
           [Vm.Shutdown]
+      | `soft_reboot ->
```

Comment:
A bit weird to have this in `on_crash_behaviour`... But we may not actually need separate functions for `on_crash_behaviour`, `on_normal_exit_behaviour` and a new one for soft reboot. Can we replace all with a single `action` function used in all cases?

Reply:
Why can't we replace the three `on_` functions with a single one?

Reply:
They are all behind the CLI so having one shoud be fine

Reply:
`on_action_behaviour` Will replace soft reboot  as a catch all and the other 2 removed.